cellpack/autopack/Analysis.py

```diff
@@ -927,6 +1010,335 @@ def PairCorrelationFunction_2D(self, x, y, S, rMax, dr):
+                            ) or ((pc1 == pc2) and (sc1 == sc2)):
+                                continue
+                            corr_df.loc[
+                                f"{pc1}_{sc1}", f"{pc2}_{sc2}"
+                            ] = self.get_spilr_correlations(
+                                all_spilr_scaled[pc1, sc1],
+                                all_spilr_scaled[pc2, sc2],
+                            )
+                            corr_df.loc[f"{pc2}_{sc2}", f"{pc1}_{sc1}"] = corr_df.loc[
+                                f"{pc1}_{sc1}", f"{pc2}_{sc2}"
+                            ]
+            df_packing = corr_df.pop("packing_id")
+            lut = dict(zip(df_packing.unique(), sns.color_palette()))
+            row_colors = df_packing.map(lut)
+            corr_df.fillna(0, inplace=True)
+            g = sns.clustermap(
+                corr_df,
+                row_colors=row_colors,
+                cbar_kws={"label": "spilr correlation"},
+            )
+            g.savefig(f"{self.output_path}spilr_correlation_{self.ingr_key}", dpi=300)
+
+        average_spilr_raw = numpy.nanmean(all_spilr_raw, axis=1)
+        average_spilr_scaled = numpy.nanmean(all_spilr_scaled, axis=1)
+        if save_plots:
+            fig_raw, ax_raw = plt.subplots()
+            fig_scaled, ax_scaled = plt.subplots()
+            for pc in range(average_spilr_raw.shape[0]):
+                g_raw = sns.heatmap(
+                    average_spilr_raw[pc],
+                    cbar=False,
+                    xticklabels=False,
+                    yticklabels=False,
+                    ax=ax_raw,
+                )
+                g_raw.set_xlabel("Angular coordinates")
+                g_raw.set_ylabel("Distance from Nuclear Surface, avg")
+                g_raw.invert_yaxis()
+
+                g_scaled = sns.heatmap(
+                    average_spilr_scaled[pc],
+                    cbar=False,
+                    xticklabels=False,
+                    yticklabels=False,
+                    ax=ax_scaled,
+                )
+                g_scaled.set_xlabel("Angular coordinates")
+                g_scaled.set_ylabel("Scaled distance from Nuclear Surface, avg")
+                g_scaled.invert_yaxis()
+
+                fig_raw = g_raw.get_figure()
+                fig_scaled = g_scaled.get_figure()
+                fig_raw.savefig(
+                    f"{save_dir}avg_heatmap_raw_{pc}_{self.ingr_key}", dpi=300
+                )
+                fig_scaled.savefig(
+                    f"{save_dir}avg_heatmap_scaled_{pc}_{self.ingr_key}",
+                    dpi=300,
+                )
+
+        return all_spilr_scaled, all_spilr_raw
+
+    @staticmethod
+    def get_spilr_correlations(spilr_list_1, spilr_list_2):
+        """
+        Calculates the Matthews correlation between spilrs
+        """
+        return matthews_corrcoef(spilr_list_1.flatten(), spilr_list_2.flatten())
+
+    @staticmethod
```

Comment:
organization: static methods should be at the top of a class ( I think before the init, but we haven't enforcing that)